module/VuFind/src/VuFind/Cover/Loader.php

```diff
@@ -697,6 +689,11 @@ protected function processImageURL($url, $cache = true)
             $tempFile = str_replace('.jpg', uniqid(), $this->localFile);
             $finalFile = $cache ? $this->localFile : $tempFile . '.jpg';
 
+            // Make sure that the cache directory exists.
+            if (!is_dir($directory = dirname($tempFile))) {
```

Comment:
Reviewing this code reveals an existing problem: there is no guarantee that $this->localFile will end with a .jpg extension, so the definition of $tempFile is not guaranteed to create a unique filename. I think we should change the definition above with:

`$tempFile = $this->localFile . uniqid();`

Reply:
I wonder why $tempFile is not handled like typical temporary files and written to TEMPDIR instead of the cache directory. Changing this would allow further optimization because the check for directory existence would only be needed when caching is enabled.